content/en/guides/deployment/deployment-koyeb.md

```diff
@@ -0,0 +1,133 @@
+---
+title: Deploy Nuxt on Koyeb Serverless Platform
+description: Deploy Nuxt on Koyeb Serverless Platform with Docker
+menu: Koyeb
+target: Server
+category: deployment
+position: 110.1
+---
+
+[Koyeb](https://www.koyeb.com) is a developer-friendly serverless platform to deploy apps globally. The platform lets you seamlessly run Docker containers, web apps, and APIs with git-based deployment, native autoscaling, a global edge network, and built-in service mesh and discovery.
+
+In this guide, we showcase how to dockerize and deploy a Nuxt application on the Koyeb platform. Koyeb allows you to deploy Docker containers from the registry of your choice. In this guide, we will push our Nuxt dockerized application on the Docker Hub and deploy it on Koyeb.
+
+## Requirements
+
+To successfully follow and complete this guide, you need:
+
+1. To have a Nuxt project to deploy. You can use the [create-nuxt-app](https://github.com/nuxt/create-nuxt-app) to create a Nuxt project and get started
+2. A [Koyeb account](https://app.koyeb.com) to deploy and run the dockerized Nuxt application
+3. Have [Docker Hub] account to push the Docker image and deploy it on Koyeb
+
+## Getting started
+
+In your Nuxt application directory run the following command to install dependencies:
+
+```bash
+# Using yarn
+yarn
+
+# Using npm
+npm install
+```
+
+Once the dependencies are installed, launch your application and ensure everything is working fine:
+
+```bash
+# Using yarn
+
+yarn dev
+
+# Using npm
+
+npm run dev
+```
+
+## Dockerize your application
+
+To Dockerize your Nuxt application, you need to create a `Dockerfile` in your project directory containing the content below:
+
+```dockerfile
+FROM node:lts
+
+WORKDIR /app
+
+COPY . ./
+RUN yarn
+
+EXPOSE 3000
+
+ENV HOST=0.0.0.0
+ENV PORT=3000
+
+RUN yarn build
+
+CMD [ "yarn", "start" ]
+```
+
+If you prefer to use `npm` instead of `yarn` set the following content in your `Dockerfile`:
+
+```dockerfile
+FROM node:lts
+
+WORKDIR /app
+
+COPY . ./
+RUN npm install
+
+EXPOSE 3000
+
+ENV HOST=0.0.0.0
+ENV PORT=3000
+
+RUN npm run build
+
+CMD [ "npm", "run", "start" ]
+```
+
+To build the Docker image execute the following command:
+
+```bash
+docker build . -t <YOUR_DOCKER_HUB_USERNAME>/my-nuxt-project
+```
+
+This command will build the Docker image with the name <YOUR_DOCKER_HUB>/my-nuxt-project. Once the build is over, you can run a container using the image locally to validate everything is working as expected running:
+
+```bash
+docker run -p 3000:3000 <YOUR_DOCKER_HUB_USERNAME>/my-nuxt-project
+```
+
+Open your browser and navigate to http://localhost:3000 to view your project landing page.
+
+## Push your Docker image on the Docker hub
+
+Since our Docker image is built and functional in our test, we can now upload it to the Docker Hub. In your terminal run the command below to push the image:
```

Comment:
Maybe worth mentioning alternative container registries like [github container registry](https://docs.github.com/en/packages/working-with-a-github-packages-registry/working-with-the-container-registry. Also any hope users can benefit from continues-deployments? (we can have a call if you would like to check possibilities)

Reply:
Of course, do you prefer the guide to use ghcr instead of Docker Hub, or mention the list of alternatives is enough? Users can benefit from CD by triggering a Koyeb deployment in their CD workflows. We are working to ease this part, in the coming weeks we will provide support of git-driven deployment which will significantly ease the flow to deploy Nuxt apps on Koyeb.

Reply:
Just mentioning possibilities would be enough (and maybe also spoiler on upcoming changes? 🔥 )

Reply:
Spoiler currently hidden in the intro 🙃 `The platform lets you seamlessly run Docker containers, web apps, and APIs with git-based deployment`